common/assertion-functions/src/main/kotlin/com/testerum/common_assertion_functions/functions/builtin/ArrayFunctions.kt

```diff
@@ -0,0 +1,108 @@
+package com.testerum.common_assertion_functions.functions.builtin
+
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.node.ArrayNode
+import com.fasterxml.jackson.databind.node.NullNode
+import com.testerum.common_assertion_functions.functions.AssertionFailedException
+import com.testerum.common_assertion_functions.functions.AssertionFunction
+import java.math.BigDecimal
+
+object ArrayFunctions {
+
+    @AssertionFunction
+    fun isArray(actualNode: JsonNode) {
+        if (actualNode is ArrayNode) {
+            return
+        }
+
+        throw AssertionFailedException("expected an Array, but got ${actualNode.toString()}")
+    }
+    
+    @AssertionFunction
+    fun isArrayOrNull(actualNode: JsonNode) {
+        if (actualNode is NullNode) {
+            return
+        }
+
+        if (actualNode is ArrayNode) {
+            return
+        }
+
+        throw AssertionFailedException("expected an Array, but got ${actualNode.toString()}")
+    }
+
+    @AssertionFunction
+    fun isNonEmptyArray(actualNode: JsonNode) {
+        if (actualNode is ArrayNode) {
```

Comment:
minor: inverting the ``if`` condition (to put the error case first and throw an exception`` would reduce the nesting

Reply:
you are right, but i have to do it everywhere
and I still need the last throw